packages/client/lib/rpc/modules/eth.ts

```diff
@@ -986,4 +988,56 @@ export class Eth {
     const block = await getBlockByOption(blockOpt, this._chain)
     return intToHex(block.transactions.length)
   }
+
+  /**
+   * Gas price oracle.
+   *
+   * Returns a suggested gas price.
+   * @returns a hex code of an integer representing the current gas price in wei.
+   */
+  async gasPrice() {
+    let gasPrice = BigInt(0)
+    const latest = await this._chain.getCanonicalHeadHeader()
+    if (this._vm !== undefined && this._vm._common.isActivatedEIP(1559)) {
+      const baseFee = latest.calcNextBaseFee()
+      let priorityFee = BigInt(0)
+      const block = await this._chain.getBlock(latest.number)
+      const blockTransactionsLength = block.transactions.length
+      if (blockTransactionsLength > 0) {
+        for (const tx of block.transactions) {
+          const maxPriorityFeePerGas = (tx as FeeMarketEIP1559Transaction).maxPriorityFeePerGas
+          priorityFee += maxPriorityFeePerGas
+        }
+      }
+      priorityFee =
+        priorityFee !== BigInt(0) ? priorityFee / BigInt(blockTransactionsLength) : BigInt(1)
+      gasPrice = baseFee + priorityFee
+    } else {
+      // For chains that don't support EIP-1559 we iterate over the last 20
+      // blocks to get an average gas price. We cap the tx lookup to 500.
```

Comment:
```suggestion
      // blocks to get an average gas price. We cap the tx lookup per block to 500.
```
Not sure if you're actually intending to cap the transaction look up in the block or total transactions examined.  If you're intending to do it as looking at a total number of transactions, you'll need to move the `break` in line 1030 after 1031.  Otherwise, it will just break from the inner loop and not the outer one.

Reply:
@acolytec3 capping it per block makes no sense (unless I'm in the wrong here), we can get the entire block body from the LevelDB immediately so just getting the avg gas price should be super fast once we got the block from the DB. (This would ofc. be different if we would have to get tx-by-hash each time)